pkg/controller/spd/indicator-plugin/resource_portrait_config_manager.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2022 The Katalyst Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package indicator_plugin
+
+import (
+	"context"
+	"encoding/json"
+
+	apiconfig "github.com/kubewharf/katalyst-api/pkg/apis/config/v1alpha1"
+	"github.com/kubewharf/katalyst-core/pkg/util/general"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/v2"
+)
+
+type ResourcePortraitConfigManager interface {
+	Refresh(ns, name string) bool
+	Filter(runtime.Object) []apiconfig.ResourcePortraitConfig
+}
+
+type resourcePortraitConfigManagerImpl struct {
+	ctx           context.Context
+	client        corev1.CoreV1Interface
+	globalConfigs *apiconfig.GlobalResourcePortraitConfiguration
+}
+
+func NewResourcePortraitConfigManager(ctx context.Context, client corev1.CoreV1Interface) ResourcePortraitConfigManager {
+	return &resourcePortraitConfigManagerImpl{
+		ctx:    ctx,
+		client: client,
+	}
+}
+
+func (m *resourcePortraitConfigManagerImpl) Refresh(ns, name string) bool {
+	cm, err := m.client.ConfigMaps(ns).Get(m.ctx, name, v1.GetOptions{ResourceVersion: "0"})
+	if err != nil {
+		klog.Errorf("[spd-resource-portrait] failed to get resource portrait config: %v", err)
+		return false
+	}
+
+	var configs []apiconfig.ResourcePortraitConfiguration
+	resourcePortraitConfig := cm.Data[ResourcePortraitIndicatorPluginName]
+	err = json.Unmarshal([]byte(resourcePortraitConfig), &configs)
```

Comment:
Using a scheme to decode might be better because it supports version control.
```
import "github.com/kubewharf/katalyst-api/pkg/client/clientset/versioned/scheme"

config := &apiconfig.GlobalResourcePortraitConfiguration{}
deserializer := scheme.Codecs.UniversalDeserializer()
_, _, err := deserializer.Decode(raw, nil, config)
```